index.js

```diff
@@ -571,6 +582,25 @@ function issecure(req, trustProxy) {
 }
 
 /**
+ * Create a new Cookie for this request.
+ *
+ * @param {Object} cookieOptions
+ * @param {Object} req
+ * @param {Boolean} [trustProxy]
+ * @return {Cookie}
+ * @private
+ */
+function createCookie(cookieOptions, req, trustProxy) {
+  cookieOptions = cookieOptions || {};
```

Comment:
@scurker What @dougwilson says above is correct.  [Here](https://gist.github.com/JoeWagner/d89ff977e47e039387f0) is a gist demonstrating the optimization pitfall